### DISCUSSION TITLE
Root cause analysis and repair items for #3277 (eBPF store APIs are not backward compatible)

### DISCUSSION THREAD

**Discussion by shankarseal** [microsoft/ebpf-for-windows#3279] [Ideas]
#3277 tracks a regression introduced in 0.14 release which is fixed in 0.14.1 by temporarily reverting the offending change #3110. This discussion is intended to root cause why the regression failed to get detected, and repair items to be implemented to avoid such regressions.

**Root Cause Analysis**:
eBPF for Windows CI pipeline does not have any backward compatibility tests for eBPF extensions.

**Repair items**:
1. Implement #3278.
2. Archive test artifacts from a release and rerun them in subsequent releases. This should catch regressions in external API contracts with extensions and also the libbpf user mode API implementation.

**Comment by Alan-Jowett** [comment]
#2747 tracks resolving this repair item.

Proposal:
Use a released version of XDP for Windows and verify it register, loads, and executes BPF programs correctly.
